packages/core/src/libraries/jwt-customizer.ts

```diff
@@ -18,42 +23,70 @@ import {
   getJwtCustomizerScripts,
   type CustomJwtDeployRequestBody,
 } from '#src/utils/custom-jwt/index.js';
+import { buildWithtypedClientResponseError } from '#src/utils/custom-jwt/index.js';
 
 import { type CloudConnectionLibrary } from './cloud-connection.js';
 
-export const createJwtCustomizerLibrary = (
-  queries: Queries,
-  logtoConfigs: LogtoConfigLibrary,
-  cloudConnection: CloudConnectionLibrary,
-  userLibrary: UserLibrary,
-  scopeLibrary: ScopeLibrary
-) => {
-  const {
-    users: { findUserById },
-    rolesScopes: { findRolesScopesByRoleIds },
-    scopes: { findScopesByIds },
-    userSsoIdentities,
-    organizations: { relations },
-  } = queries;
-  const { findUserRoles } = userLibrary;
-  const { attachResourceToScopes } = scopeLibrary;
-  const { getJwtCustomizers } = logtoConfigs;
+export class JwtCustomizerLibrary {
+  // Convert errors to WithTyped client response error to share the error handling logic.
+  static async runScriptInLocalVm(data: CustomJwtFetcher) {
+    try {
+      const payload =
+        data.tokenType === LogtoJwtTokenKeyType.AccessToken
+          ? pick(data, 'token', 'context', 'environmentVariables')
+          : pick(data, 'token', 'environmentVariables');
+      const result = await runScriptFunctionInLocalVm(data.script, 'getCustomJwtClaims', payload);
+
+      // If the `result` is not a record, we cannot merge it to the existing token payload.
+      return z.record(z.unknown()).parse(result);
+    } catch (error: unknown) {
+      // Assuming we only use zod for request body validation
+      if (error instanceof ZodError) {
+        const { errors } = error;
+        throw buildWithtypedClientResponseError(
+          {
+            message: 'Invalid input',
+            errors,
+          },
+          400
+        );
+      }
+
+      throw buildWithtypedClientResponseError(
+        buildErrorResponse(error),
+        error instanceof SyntaxError || error instanceof TypeError ? 422 : 500
+      );
+    }
```

Comment:
why do we need to build a `withtyped` error here?

Reply:
Align the thrown errors with the error we got from Azure Functions and Cloudflare Workers, so that we can reuse the error handling and display logic.